x/coin.go

```diff
@@ -60,7 +60,7 @@ func (c Coin) ID() string {
 //   c.Add(o.Negative())
 func (c Coin) Add(o Coin) (Coin, error) {
 	if !c.SameType(o) {
-		err := ErrInvalidCurrency(c.Ticker, o.Ticker)
+		err := InvalidCurrencyErr.New("%s vs %s", c.Ticker, o.Ticker)
```

Comment:
Please use `Newf` as above, but thank you for adding this context.

I would change the message to `Adding %s to %s` to make it clearer....
`InvalidCurrencyErr.Newf("Adding %s to %s", c.Ticker, o.Ticker)`